### DISCUSSION TITLE
不支持ipv6的智能家居设备如何与prefer_ipv6共存

### DISCUSSION THREAD

**Discussion by uvswifft** [IrineSistiana/mosdns#741] [Q&A]
### 在提交之前，请确认

- [X] 我已经尝试搜索过 Issue ，但没有找到相关问题。
- [X] 我正在使用最新的 mosdns 版本(或者最新的 commit)，问题依旧存在。
- [X] 我仔细看过 wiki 后仍然无法自行解决该问题。
- [X] 我非常确定这是 mosdns 核心的问题。(如果是通过第三方衍生软件使用 mosdns 核心，不确定问题源头时，请先向衍生软件开发者提交问题。)

### mosdns 版本

v5.2.1

### 操作系统

docker

### Bug 描述和复现步骤

一部分智能家居设备由于不支持ipv6，所以只会发送A记录请求，但如果mosdns开启prefer_ipv6，mosdns会同时发出AAAA记录请求，这时如果查询到的域名解析是双栈ip，则会屏蔽A记录，对于智能家居设备而言就是请求无响应，导致连不上服务器。

这个问题对于其它类似单栈设备应该是普遍现象，单ipv4的智能家居设备应该还是挺多的。

### 使用的配置文件

```yaml
。。。
```

**Comment by IrineSistiana** [comment]
> prefer_ipv6存在设计缺陷

所以你的方案是？

**Comment by uvswifft** [comment]
相关域名单独解析，不设置prefer，这玩意好像也没啥其它靠谱方法解决

**Comment by uvswifft** [comment]
说起来这个，A/AAAA记录都可以对响应ip进行判断在不在特定IP内，不知道https记录的响应能不能进行

**Comment by emptyteeth** [comment]
是不是可以用client_ip匹配器来决定要不要执行prefer ipv6?

if ! client_ip 192.168.4.0/24 then prefer ipv6
或者只对v6 client执行
if client_ip ::/0 then prefer ipv6

**Comment by sensec** [comment]
定义一个智能家居的IP表，根据 client_ip 是否在表内决定 prefer_ipv4/6。
